### ISSUE TITLE
variable (in xxx_data) in attributes of tag

### ISSUE THREAD

**Issue by rogerburton** [derekmc/lisp-markup#1]
excuse me for my bad English
I can't find a way to insert a variable in a template at the level of a tag's attributes

`      var main_data = {
          prime_list: [2,3,5,7,11,13,17,19]
      };
      var asdf_data={zut: "test me !"}`
[...]
` (SECTION asdf (h2 ASDF) (h3 TEST (@ zut))
        (p (FOR $x (1 10 0.7)
          (= ($z (@ zut))
          (input 
            {type text value $z} $x)))))`

(h3 TEST (@ zut)) is OK
but I did not find the syntax for
{type text value $z} 

your code is very very clever and the Lisp syntax is really elegant/efficient for this use
Thanks
Roger Burton

**Comment by derekmc** [comment]
yes, you are certainly putting this old library through its paces.

I can't really remember all the details of how I implemented macros, so the issue may arise from the assignment nested inside the "FOR" macro.

While this library was used in one production project(draw15 in google play store), which you can see if you download the apk and unzip it, I have since moved to simpler templating approaches, because lisp-markup was getting really hard to debug.

Here is a smattering of various simpler templating approaches I have experimented with since then.
especially using javascript's template strings:
- [Template Strings Example](https://codepen.io/math3737/pen/abWrxgy?editors=0010)
- [Tag Functions](https://github.com/derekmc/tagfuncs/blob/master/test.js)  (See also ["crel"](https://github.com/KoryNunn/crel))
- [ezhtml](https://gitlab.com/derekmc/ezhtml/-/blob/master/example/index.html)
- [brackets](https://gitlab.com/derekmc/derekmc.gitlab.io/-/blob/main/projects/brackets/test.js) Uses significant whitespace.
- [json-html](https://gitlab.com/derekmc/derekmc.gitlab.io/-/blob/main/projects/json-markup/jsonhtml.js) Also available on npm, may not be working.
- [ls "framework"](https://gitlab.com/derekmc/derekmc.gitlab.io/-/blob/main/sandbox/lsframework/index.js) (really just a single one line helper function for use inside template string literals, etc.


I must admit, nothing is quite as beautiful as the lisp syntax, but implementing syntax checking and constructive error messages proved difficult, not to mention the complexity of the macro implementation made it a pain in the butt to debug.

After much experimentation, my conclusion has been that template inheritance is the most useful templating feature, unfortunately there do not seem to be good implementations of template inheritance or similar in client-side templating engines.  I have resorted to approaches like using jinja and then crawling the pages.  I am frankly quite opposed to transpilation and try to avoid it as much as possible, which is not easy in the javascript world.  So while there may be approaches for running templating languages with template inheritance, none of them were quite satisfactory to me.

There is an implementation of client-side templating in this game I am a little bit fond of, it allows you to amalgamate many different pages into a single document and rendered the desired page [Peasant's MMT Quest](https://www.newgrounds.com/portal/view/758939)

Happy templating!

PS I may not get around to fixing this library, but if you share more info to help me reproduce, it will at least give me something to take a crack at.
